clouddriver-docker/src/main/groovy/com/netflix/spinnaker/clouddriver/docker/registry/api/v2/client/DockerRegistryClient.groovy

```diff
@@ -351,9 +351,31 @@ class DockerRegistryClient {
         }
       } catch (RetrofitError error) {
         if (error.response?.status == 401) {
-          dockerToken = tokenService.getToken(target, error.response.headers)
-          token = "Bearer ${dockerToken.bearer_token ?: dockerToken.token}"
-          response = withToken(token)
+          String authenticateHeader = null
+
+          error.response.headers.forEach { header ->
+            if (header.name.equalsIgnoreCase("www-authenticate")) {
```

Comment:
This bearer token specific code should probably live in the `DockerBearerTokenService`